pkg/admin.go

```diff
@@ -81,35 +81,45 @@ func (r *Registry) VendorClaim(vendorID string, orgID string, orgName string, or
 	if vendor == nil {
 		return nil, fmt.Errorf("vendor not found (id=%s)", vendorID)
 	}
-	certificates := vendor.GetActiveCertificates()
-	if len(certificates) == 0 {
-		return nil, fmt.Errorf("vendor has no active certificates (id = %s)", vendorID)
-	}
 
-	// If no keys are supplied, make there's a key in the crypto module for the organisation
-	if orgKeys == nil || len(orgKeys) == 0 {
-		logrus.Infof("No keys specified for organisation (id = %s). Keys will be generated or loaded from crypto module.", orgID)
+	// If no keys are supplied, make sure there's a key in the crypto module for the organisation
+	if len(orgKeys) == 0 {
+		logrus.Infof("No keys specified for organisation (id=%s). Keys will be generated or loaded from crypto module.", orgID)
 		_, err := r.loadOrGenerateKey(orgID)
 		if err != nil {
 			return nil, err
 		}
 	}
 
-	certificate, err := r.createAndSubmitCSR(func() (x509.CertificateRequest, error) {
-		return cert.OrganisationCertificateRequest(vendor.Name, orgID, orgName, vendor.Domain)
-	}, types.LegalEntity{URI: orgID}, types.LegalEntity{URI: vendorID}, crypto.CertificateProfile{
-		IsCA:         true,
-		KeyUsage:     x509.KeyUsageDigitalSignature | x509.KeyUsageKeyEncipherment | x509.KeyUsageDataEncipherment,
-		NumDaysValid: r.Config.OrganisationCertificateValidity,
-	})
-	if err != nil {
-		return nil, errors2.Wrap(err, ErrCertificateIssue.Error())
-	}
+	certificates := vendor.GetActiveCertificates()
+	if len(certificates) > 0 {
+		// If the vendor has certificates, it means it has (should have) a CA certificate which can issue a certificate to the new org
+		certificate, err := r.createAndSubmitCSR(func() (x509.CertificateRequest, error) {
+			return cert.OrganisationCertificateRequest(vendor.Name, orgID, orgName, vendor.Domain)
+		}, types.LegalEntity{URI: orgID}, types.LegalEntity{URI: vendorID}, crypto.CertificateProfile{
+			IsCA:         true,
+			KeyUsage:     x509.KeyUsageDigitalSignature | x509.KeyUsageKeyEncipherment | x509.KeyUsageDataEncipherment,
+			NumDaysValid: r.Config.OrganisationCertificateValidity,
+		})
+		if err != nil {
+			return nil, errors2.Wrap(err, ErrCertificateIssue.Error())
+		}
 
-	jwkAsMap, err := certToJwkMap(certificate, cert.OrganisationCertificate)
-	orgKeys = append(orgKeys, jwkAsMap)
-	if err != nil {
-		return nil, errors2.Wrap(err, ErrJWKConstruction.Error())
+		jwkAsMap, err := certToJwkMap(certificate, cert.OrganisationCertificate)
+		orgKeys = append(orgKeys, jwkAsMap)
+		if err != nil {
+			return nil, errors2.Wrap(err, ErrJWKConstruction.Error())
+		}
+	} else {
+		// Vendor has no certificates, we just make sure the org has a plain JWK without X.509 certificate, either
+		// provided or freshly generated.
```

Comment:
Add a comment that this should be removed when all vendors have certificates.

Reply:
Fixed, specified in the same words as in the comment as below (to ease searching for what to remove).